_rules/descriptive-link-5effbb.md

```diff
@@ -0,0 +1,292 @@
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<a href="#desc">More</a>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Failed Example 2
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<div role="link" tabindex="0" onclick="document.location+='#main'">More</div>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 3
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" x="0" y="0">
+	<a href="#main">
+		<text x="20" y="20">
+			Go
+		</text>
+	</a>
+</svg>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 4
+
+The [accessible name][] (from the link's text) does not describe the purpose of the link. The other information available on the page is not [programmatically determined link context][] because it is on a different paragraph.
+
+```html
+<p>
+	The W3C held a workshop on June 9-10, 2005 at DERI Innsbruck (Austria), to gather information about potential
+	standardization work on Semantics in Web Services.
+</p>
+
+<p><a href="https://www.w3.org/2005/04/FSWS/workshop-report.html">Workshop</a></p>
+```
+
+#### Failed Example 5
+
+The [accessible name][] (from the link's text) does not describe the purpose of the link. The other information available on the page is not [programmatically determined link context][] because it is outside the list where the links are.
+
+```html
+<p style="font-weight: bold">Ulysses</p>
+<ul>
+	<li><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm"> HTML </a></li>
+	<li>
+		<a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73">
+			EPUB
+		</a>
+	</li>
+	<li><a href="https://www.gutenberg.org/files/4300/4300-0.txt"> Plain text </a></li>
+</ul>
+```
+
+#### Failed Example 6
+
+The [accessible name][] (from the link's text) does not describe the purpose of the link. The other information available on the page is not [programmatically determined link context][] because it not available on the same cell of the link or in a header cell for that cell.
+
+```html
+<table>
+	<tr>
+		<th colspan="3">Books</th>
+	</tr>
+	<tr>
+		<td>Ulysses</td>
+		<td><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">Download</a></td>
+		<td>1.61MB</td>
+	</tr>
+</table>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+An `a` element with its [semantic role][] changed from `link` to another role.
+
+```html
+<a href="http://www.w3.org/WAI" role="button">Web Accessibility Initiative (WAI)</a>
```

Comment:
That's typically breaking the assumption "all link have a role of link", and I don't think we should have example that go against assumptions. In that case, I think the assumption should go away.